### PR TITLE
feat(workflow): Adding has_sessions flag to Project model

### PR DESCRIPTION
**Pull request by iProgramStuff**
Adding the `has_sessions` flag to the Project model.

The intention here is to return it with the project serializer if you pass an expand parameter. If `has_sessions` is true, we'll use it as-is, but if it is false we'll run a query to see if the project had sessions and cache the result for an hour (so we run this query max once per project per hour).

We'll only use this expand parameter on the issue list page for the time being.

### CODE REVIEW

**Review by wedamija — COMMENTED** [review]
Could you check that this flag is currently unset for all projects?

**Comment by iProgramStuff** [comment]
Verified ```select * from sentry_project where (flags >> 4) > 0 ``` and
```select * from sentry_project where flags>15``` return no results

File: src/sentry/migrations/0223_has_sessions_flag.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = True
```
Comment (wedamija) [inline src/sentry/migrations/0223_has_sessions_flag.py:18]: Doesn't need to be dangerous, since it's not doing anything.

**Review by wedamija — APPROVED** [review]

**Review by wedamija — APPROVED** [review]

**Comment by matejminar** [comment]
just fyi, @ahmedetefy added recently `sessionStats` to projects endpoint
Calling
`/api/0/organizations/my-org/projects/?statsPeriod=24h&transactionStats=1&sessionStats=1`
Will return among other things something like this:
`"sessionStats":{"currentCrashFreeRate":99.83865552081562,"previousCrashFreeRate":99.80525044792397,"hasHealthData":true}`

**Comment by ahmedetefy** [comment]
> just fyi, @ahmedetefy added recently sessionStats to projects endpoint
Calling
/api/0/organizations/my-org/projects/?statsPeriod=24h&transactionStats=1&sessionStats=1
Will return among other things something like this:
"sessionStats":{"currentCrashFreeRate":99.83865552081562,"previousCrashFreeRate":99.80525044792397,"hasHealthData":true}

Yeah might be unnecessary to calculate `hasHealthData` in that dict and just rely on `has_sessions` flag @iProgramStuff  is adding

**Pull request merged**